### ISSUE TITLE
Experiments on ImageNet with R18 or R50?

### ISSUE THREAD

**Issue by jiequancui** [davda54/sam#40]
Hi, is there anyone who has reproduced the results reported by the paper on ImageNet with ResNet-18 or ResNet-50?

**Comment by shuo-ouyang** [comment]
SAM got an approximate 0.3% top-1 validation accuracy improvement compared with vanilla SGD in my experiments, but I use MXNet...

**Comment by jiequancui** [comment]
Many thanks, I find that fixing batch normalization at the first forward is a key factor.

**Comment by shuo-ouyang** [comment]
Well, I do not fix any BN layers during the training process. However, the model diverges when I save the running mean and var at the first pass and restore them at the second pass. Could you give more detail about fixing BN? Thanks very much!

**Comment by jiequancui** [comment]
Yes, I have met the same problem before.  And I do the following changes like in adversarial trianing:

 for the first forward backwar:
             for name, module in model.named_modules:
                   if isinstance(module, torch.nn.BatchNorm2d):
                      module.eval()
              out = model(input)
              loss = crossentropy_smooth(out, target)
              loss.backward()

  for the closure function:
           for name, module in model.named_modules:
                   if isinstance(module, torch.nn.BatchNorm2d):
                      module.train()
              out = model(input)
              loss = crossentropy_smooth(out, target)
              loss.backward()
              return loss

**Comment by shuo-ouyang** [comment]
Thanks for your valuable advice. I will re-run my experiments and check the result.

**Comment by jiequancui** [comment]
Many thank you for the open-sourced code.
